ts/features/bonus/bpd/store/reducers/__test__/combiner.test.ts

```diff
@@ -0,0 +1,338 @@
+      const visiblePeriods = bpdPeriodsAmountWalletVisibleSelector.resultFunc(
+        pot.some([
+          closedPeriodZeroAmount,
+          activePeriodAmount,
+          inactivePeriodA,
+          inactivePeriodB,
+          inactivePeriodC
+        ]),
+        remoteReady(true)
+      );
+
+      expect(pot.isSome(visiblePeriods)).toBeTruthy();
+      if (pot.isSome(visiblePeriods)) {
+        expect(visiblePeriods.value.length).toBe(1);
+        expect(visiblePeriods.value[0].period.awardPeriodId).toBe(
+          activePeriodAmount.period.awardPeriodId
+        );
+      }
+    }
+  );
+
+  it(
+    "with multiple inactive period, an active period and a closed period (with amount > zero) " +
+      "should return the active period and the closed period",
+    () => {
+      const visiblePeriods = bpdPeriodsAmountWalletVisibleSelector.resultFunc(
+        pot.some([
+          closedPeriodWithAmount,
+          activePeriodAmount,
+          inactivePeriodA,
+          inactivePeriodB,
+          inactivePeriodC
+        ]),
+        remoteReady(true)
+      );
+
+      expect(pot.isSome(visiblePeriods)).toBeTruthy();
+      if (pot.isSome(visiblePeriods)) {
+        expect(visiblePeriods.value.length).toBe(2);
+        expect(visiblePeriods.value[0].period.awardPeriodId).toBe(
+          closedPeriodWithAmount.period.awardPeriodId
+        );
+        expect(visiblePeriods.value[1].period.awardPeriodId).toBe(
+          activePeriodAmount.period.awardPeriodId
+        );
+      }
+    }
+  );
+
+  it(
+    "with multiple inactive period and a closed period (with amount > zero) " +
+      "should return the closed period",
+    () => {
+      const visiblePeriods = bpdPeriodsAmountWalletVisibleSelector.resultFunc(
+        pot.some([
+          closedPeriodWithAmount,
+          inactivePeriodA,
+          inactivePeriodB,
+          inactivePeriodC
+        ]),
+        remoteReady(true)
+      );
+
+      expect(pot.isSome(visiblePeriods)).toBeTruthy();
+      if (pot.isSome(visiblePeriods)) {
+        expect(visiblePeriods.value.length).toBe(1);
+        expect(visiblePeriods.value[0].period.awardPeriodId).toBe(
+          closedPeriodWithAmount.period.awardPeriodId
+        );
+      }
+    }
+  );
+  it(
+    "with multiple inactive period and a closed period (with amount zero) " +
+      "should return no periods",
+    () => {
+      const visiblePeriods = bpdPeriodsAmountWalletVisibleSelector.resultFunc(
+        pot.some([
+          closedPeriodZeroAmount,
+          inactivePeriodA,
+          inactivePeriodB,
+          inactivePeriodC
+        ]),
+        remoteReady(true)
+      );
+
+      expect(pot.isSome(visiblePeriods)).toBeTruthy();
+      if (pot.isSome(visiblePeriods)) {
+        expect(visiblePeriods.value.length).toBe(0);
+      }
+    }
+  );
+});
+
+describe("test bpdPeriodsAmountWalletVisibleSelector when bpd is disabled", () => {
+  it("one inactive period should return one inactive period", () => {
```

Comment:
```suggestion
  it("one inactive period should return no periods", () => {
```

Reply:
When there is one inactive (future) period, that period should be returned, following the new logic.